src/network/networkd-json.c

```diff
@@ -1161,6 +1162,141 @@ static int dns_misc_build_json(Link *link, JsonVariant **ret) {
         return r;
 }
 
+static int dhcp_server_offered_leases_build_json(Link *link, JsonVariant **ret) {
+        _cleanup_(json_variant_unrefp) JsonVariant *elements = NULL;
+        int r;
+        DHCPLease *lease;
+
+        assert(link);
+        assert(ret);
+
+        if (!link->dhcp_server) {
+                *ret = NULL;
+                return 0;
+        }
+
+        if (hashmap_isempty(link->dhcp_server->bound_leases_by_client_id)) {
+                *ret = NULL;
+                return 0;
+        }
+
+        HASHMAP_FOREACH(lease, link->dhcp_server->bound_leases_by_client_id) {
+                _cleanup_(json_variant_unrefp) JsonVariant *e = NULL;
+                _cleanup_free_ char *id = NULL;
+                struct in_addr address = { .s_addr = lease->address };
+
+                r = sd_dhcp_client_id_to_string(lease->client_id.data, lease->client_id.length, &id);
+                if (r < 0) {
+                        *ret = NULL;
+                        return 0;
+                }
+
+                r = json_build(&e,JSON_BUILD_OBJECT(
+                        JSON_BUILD_PAIR_STRING("ClientId", id),
+                        JSON_BUILD_PAIR_IN4_ADDR_NON_NULL("Address", &address),
+                        JSON_BUILD_PAIR_UNSIGNED_NON_ZERO("Expiration", lease->expiration)));
```

Comment:
Please use +16ch indentation for long function arguments.

Reply:
Done.